pandas/tests/extension/base/groupby.py

```diff
@@ -33,6 +33,15 @@ def test_groupby_extension_agg(self, as_index, data_for_grouping):
             expected = expected.reset_index()
             self.assert_frame_equal(result, expected)
 
+    def test_groupby_agg_extension(self, data_for_grouping):
+        # GH#38980 groupby agg on extension type fails for non-numeric types
+        df = pd.DataFrame({"A": [1, 1, 2, 2, 3, 3, 1, 4], "B": data_for_grouping})
+        result = df.groupby("A").agg({"B": "first"}).B.array
+
+        expected = df["B"].iloc[[0, 2, 4, 7]].array
```

Comment:
expected should be a dataframe here. pls explicity construct it and use assert_frame_equal